toolkit/scripts/check_signatures.py

```diff
@@ -20,20 +22,65 @@ def getSignature(fileName) -> str:
     return sha256sum.hexdigest()
 
-def check_folder(path):
+def find_file(path, filename) -> Optional[str]:
+    return_value : Optional[str] = None
+    for matching_file in Path(path).glob(f"**/{filename}"):
+        if os.path.exists(matching_file):
+            if return_value is not None:
+                print(f"ERROR: detected multiple {filename}: [{matching_files}]")
+                return None
+            
+            return_value = matching_file
+
+    return return_value
+
+def find_name_of_all_spec_and_signatures_json_pairs(path: str) -> List[str]:
+    names: List[str] = []
+    # Search for all spec files (XXX.spec)
+    for spec_path in Path(path).glob("*.spec"):
+        if os.path.exists(spec_path):
+            name = Path(spec_path).stem
+            signature_path = os.path.join(path, f"{name}.signatures.json")
+            if os.path.exists(signature_path):
+                # If there is a matching signature file (XXX.signatures.json),
+                # add it to list
+                names.append(name)
+
+    return names
+
+def find_spec_folder_ancestor(path: str) -> Optional[str]:
+    # Assume that spec/signatures.json files are only found in
+    # SPECS/XX and SPECS-EXTENDED/XX.  Find an ancestor of path
+    # that adheres to this assuption.  Return None if not found.
+    regex = f".*SPECS(-EXTENDED)?{os.sep}[^{os.sep}]+"
+    matching_path = re.search(regex, path)
+    if matching_path is not None:
+        return matching_path.group(0)
+    return None
+
+def check_folder(folder):
     signatures_correct = True
-    for signature_path in Path(path).glob("*.signatures.json"):
+
+    # get SPECS/XX or SPECS-EXTENED/XX ancestor of input
+    path = find_spec_folder_ancestor(folder)
+    if path is None:
+        # no spec/signature files found in path or its ancestors
+        return signatures_correct
+
+    for name in find_name_of_all_spec_and_signatures_json_pairs(path):
+        signature_path = os.path.join(path, f"{name}.signatures.json")
         with open(signature_path, "r") as f:
             signatures_json = json.load(f)
             for file_to_check, expected_signature in signatures_json["Signatures"].items():
-                path_to_check = join(path, file_to_check)
-                if Path(path_to_check).is_file():
-                    actual_signature = getSignature(path_to_check)
+                file_to_check_path = find_file(path, file_to_check)
+                if file_to_check_path is not None:
+                    actual_signature = getSignature(file_to_check_path)
                     if actual_signature != expected_signature:
                         print(f"ERROR: detected a mismatched signature for {file_to_check}, expected [{expected_signature}] does not equal actual [{actual_signature}]")
                         signatures_correct = False
                 else:
                     print(f"{file_to_check} is not found in CBL-Mariner, build to verify signature")
```

Comment:
Highly unlikely but if `find_file` finds more than one file (and returns `None` as a result of that), this message will make things confusing. I'd suggest we print file-related error messages either from here or from `find_file`, so that they are consistent.